framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java

```diff
@@ -660,6 +631,57 @@ public String toString() {
         }
     }
 
+    /**
+     * Comparator for CheckerMessage objects. Sort by position at which the error will be printed,
+     * then by kind of message, and finally by the message string. If different checkers have
+     * different messages, sort the messages by the order in which the checkers run.
```

Comment:
Sorry, I still don't understand.

Is the second sentence something that happens after the "finally" of the first sentence?  (That would be confusing, since I assumed "finally" was the last thing.)

Or is the second sentence the very first thing that happens?  (In that case, I would have expected it to appear before the other sentence.  Also, since different checkers have different messages, then they are already sorted by the first sentence, so I don't see that the second sentence would have any effect.)

If it's not one of those, does it happen in between two of the previous three criteria?

Could you please clarify my confusion?

Reply:
I simplified the list to hopefully make it clearer. Does it help?